src/main/java/com/plivo/api/models/message/MessageCreator.java

```diff
@@ -22,19 +22,21 @@
   private final String text;
   @JsonProperty("powerpack_uuid")
   private String powerpackUUID;
-  private MessageType type = MessageType.SMS;
+  private MessageType type = null;
   private URL url = null;
   private String method = "POST";
   private Boolean log = null;
   private Boolean trackable = null;
+  private  String[] media_urls = null;
+
 
   /**
    * @param source The phone number that will be shown as the sender ID.
    * @param destination The numbers to which the message will be sent.
    * @param text The text message that will be sent.
    */
   MessageCreator(String source, List<String> destination, String text) {
-    if (!Utils.allNotNull(source, destination, text)) {
+    if (!Utils.allNotNull(source, destination)) {
       throw new IllegalArgumentException("source, destination and text must not be null");
```

Comment:
Change error string as well. You are not checking for text == null here, but error says you are checking text